modules/jvm/v9/artifacts/opt/jboss/container/java/jvm/java-default-options

```diff
@@ -0,0 +1,146 @@
+#!/bin/sh
+# =================================================================
+# Detect whether running in a container and set appropriate options
+# for limiting Java VM resources
+#
+# Usage: JAVA_OPTS="$(java-default-options.sh)"
+
+# Env Vars respected:
+
+# JAVA_OPTS: Checked for already set options
+# JAVA_INITIAL_MEM_RATIO: Ratio of maximum memory to use for initial heap size
+#                         (i.e. -Xms).  Defaults to 25 (i.e -Xms=-Xmx/4).
+# JAVA_MAX_INITIAL_MEM: The maximum value of the initial heap size, defaults to 4G.
+# JAVA_MAX_MEM_RATIO: Ratio use to calculate a default maximum Memory, in percent.
+#                     E.g. the default value "50" implies that 50% of the Memory
+#                     given to the container is used as the maximum heap memory with
+#                     '-Xmx'. It is a heuristic and should be better backed up with real
+#                     experiments and measurements.
+#                     For a good overviews what tuning options are available -->
+#                             https://youtu.be/Vt4G-pHXfs4
+#                             https://www.youtube.com/watch?v=w1rZOY5gbvk
+#                             https://vimeo.com/album/4133413/video/181900266
+# Also note that heap is only a small portion of the memory used by a JVM. There are lot
+# of other memory areas (metadata, thread, code cache, ...) which addes to the overall
+# size. There is no easy solution for this, 50% seems to be are reasonable compromise.
+# However, when your container gets killed because of an OOM, then you should tune
+# the absolute values
+#
+
+__DEFAULT_JAVA_MAX_MEM_RATIO=50
+__DEFAULT_JAVA_INITIAL_MEM_RATIO=25
+__DEFAULT_JAVA_MAX_INITIAL_MEM=4096
+
+# stubs for jvm specific overrides
+jvm_specific_options() {
+  :
+}
+
+jvm_specific_diagnostics() {
+  :
+}
+
+# Include overridden jvm_specific_*() functions
+if [ -f "${JBOSS_CONTAINER_OPENJDK_JDK_MODULE}/jvm-options" ]; then
+  source "${JBOSS_CONTAINER_OPENJDK_JDK_MODULE}/jvm-options"
+fi
+
+initialize_container_limits() {
+  # we can't run without limits
+  source ${JBOSS_CONTAINER_JAVA_JVM_MODULE}/container-limits
+}
+
+# Check for memory options and calculate a sane default if not given
+max_memory() {
+  # Check if explicitly disabled
+  if [ "x$JAVA_MAX_MEM_RATIO" = "x0" ]; then
+    return
+  fi
+
+  # Check for the 'real memory size' and calculate mx from a ratio
+  # given (default is 50%)
+  if [ "x$CONTAINER_MAX_MEMORY" != x ]; then
+    local max_mem="${CONTAINER_MAX_MEMORY}"
+    local ratio=${JAVA_MAX_MEM_RATIO:-${__DEFAULT_JAVA_MAX_MEM_RATIO}}
+    local mx=$(echo "${max_mem} ${ratio} 1048576" | awk '{printf "%d\n" , ($1*$2)/(100*$3) + 0.5}')
+    echo "-Xmx${mx}m"
+  fi
+}
```

Comment:
This should get replaced by setting `-XX:MaxRAMPercentage=80` or if we really need 1-to-1 compat to `50`.